From 0e003e6f7a2ac983c1972c630c280e045cb36340 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <sarishar@qti.qualcomm.com>
Date: Sat, 12 Jul 2025 12:23:10 +0530
Subject: UPSTREAM:DP: wifi:ath12k: Decrement TID on cleanup for error

Currently, TID is not decremented before peer cleanup, during error
handling path of ath12k_dp_rx_peer_frag_setup(). This could lead to
out-of-bounds access in peer->rx_tid[].

Hence, add a decrement operation for TID, before peer cleanup to
ensures proper cleanup and prevents out-of-bounds access issues when
the RX peer frag setup fails.

Signed-off-by: Sarika Sharma <sarishar@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp.c b/drivers/net/wireless/ath/ath12k/dp.c
index 8d6f4e6..91f089c 100644
--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -166,7 +166,7 @@ int ath12k_dp_peer_setup(struct ath12k *ar, struct ath12k_link_vif *arvif, const
 peer_clean:
 	spin_lock_bh(&dp->dp_lock);
 
-	for (; tid >= 0; tid--)
+	for (tid--; tid >= 0; tid--)
 		ath12k_dp_arch_rx_peer_tid_delete(ab->dp, ar, peer, tid);
 
 	spin_unlock_bh(&dp->dp_lock);
-- 
2.34.1

